README.md

```diff
@@ -62,15 +79,71 @@ check` before proceeding to debug using `bin/webpack`.
 
-## Documentation
-
-### Nomenclature
+## <a name="documentation"></a>Documentation
 
+### <a name="documentation-nomenclature"></a>Nomenclature
 - **Course** consists of a UCAS provider code and a UCAS course code. In our system, this is represented by the `ucas_provider_code` and `ucas_course_code` on the `ApplicationChoice` model
 - **Course Choice** is the course plus a training location code, in our system represented by `ucas_provider_code`, `ucas_course_code`, `ucas_location_code`
 
-### Domain model
+### <a name="documentation-domain-model"></a>Domain Model
 
 ![The domain model for this application](docs/domain-model.png)
 
 Regenerate this diagram with `bundle exec erd`.
+
+### <a name="documentation-env-vars"></a>Environment Variables
+
+**NOTE: Environment variables should not start with *endpoint*, *input*, *secret*, or *securefile* (irrespective of capitalisation) due to them being protected variable names within the Azure DevOps environment.** If this cannot be avoided variable mapping will have to be used, but wherever possible it is simpler not to use these protected names.
+
+Environment variables have to be defined in several places depending upon where they are required, some are common to both local development and the Azure hosted deployment, while others are specific to the envirionments they relate, all of which are described below
+
+#### <a name="documentation-env-vars-dockerfile"></a>Dockerfile
+
+If an environment variable is required during the Docker image build, like for example in the Rails asset compilation process, these should be defined in the Dockerfile in the following format.
+
+`ENV VARIABLE_NAME=Value`
+
+When defining variables in the Dockerfile use dummy values for security reasons since they will be committed to Git and they will ultimately be overriden by docker-compose when you launch a container using the image.
+
+#### <a name="documentation-env-vars-local-dev"></a>Local Development
+
+If an environment variable is required for use in the local development environment it should be declared in the .env file in the following format 
+
+`VARIABLE_NAME=Value`
+
+The [.env.example](./.env.example) file contains the essential environment variables that must exist for local development builds to succeed.
+
+#### <a name="documentation-env-vars-docker-compose"></a>Docker Compose
+
+For docker compose to make the necessary environment variables available in the container at run time they must be declared in the relevant docker-compose.yaml file, of which there are three.
+
+* [docker-compose.yml](./docker-compose.yml) - Variables that are required for local dev and the docker image build phase in azure only should be defined the *environment* section in this file. This will in general only be for the database.
+* [docker-compose.override.yml](./docker-compose.override.yml) - This file is used exclusively for local development and imports variables from the .env file so no further changes are required here.
```

Comment:
`docker-compose`'s behaviour with `.env` is a bit funky. From the docs I think it uses `.env` to resolve environment variables inside the `docker-compose.yml` file itself (https://docs.docker.com/compose/environment-variables/#the-env-file). It's a happy accident that `.env` is copied inside the container where Rails can pick up its values. Could we just say "this file is used exclusively for local development"?

Reply:
```suggestion
* [docker-compose.override.yml](./docker-compose.override.yml) - This file is used exclusively for local development. No further changes are required here.
```